nexus/auth/src/authn/mod.rs

```diff
@@ -391,6 +417,11 @@ enum Kind {
 pub struct Details {
     /// the actor performing the request
     actor: Actor,
+    /// When the device token expires. Present only when authenticating via
+    /// a device token. This is a slightly awkward fit but is included here
+    /// because we need to use this to clamp the expiration time when device
+    /// tokens are confirmed using an existing device token.
+    device_token_expiration: Option<DateTime<Utc>>,
```

Comment:
This is the sad part. I considered doing a proper `scheme: AuthnScheme` struct here like `AuthnScheme::DeviceToken { time_expires: Option<DateTime<Utc>> }` but I think that makes the abstraction-leaking here even worse. We don't want app-layer code to be doing conditional logic based on the authn scheme. Authz policy is generally meant to cover that. The device token confirmation logic in this PR is a special case.